pkg/okteto/kubetoken/cache.go

```diff
@@ -0,0 +1,122 @@
+package kubetoken
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/okteto/okteto/pkg/log"
+	authenticationv1 "k8s.io/api/authentication/v1"
+)
+
+const cacheIsCorrupted = "cache is corrupted"
+
+type stringStore interface {
+	Get() ([]byte, error)
+	Set([]byte) error
+}
+
+type Cache struct {
+	StringStore stringStore
+	Now         func() time.Time
+	Debug       func(string, ...interface{})
+}
+
+func NewCache(fileName string) *Cache {
+	return &Cache{
+		StringStore: NewFileByteStore(fileName),
+		Now:         time.Now,
+		Debug:       log.Debugf,
+	}
+}
+
+type storeRegister struct {
+	ContextName string                        `json:"context"`
+	Namespace   string                        `json:"namespace"`
+	Token       authenticationv1.TokenRequest `json:"token"`
+}
+
+func (c *Cache) read() ([]storeRegister, error) {
+	contents, err := c.StringStore.Get()
+	if err != nil {
+		return nil, err
+	}
+
+	if len(contents) == 0 {
+		return []storeRegister{}, nil
+	}
+
+	var store []storeRegister
+
+	if err := json.Unmarshal(contents, &store); err != nil {
+		return nil, fmt.Errorf(cacheIsCorrupted)
+	}
+
+	return store, nil
+}
+
+func (c *Cache) Get(contextName, namespace string) (string, error) {
+	store, err := c.read()
+	if err != nil {
+		return "", err
+	}
+
+	for _, register := range store {
+		if register.ContextName == contextName && register.Namespace == namespace {
+			now := c.Now()
+			if register.Token.Status.ExpirationTimestamp.Time.After(now) {
+				tokenString, _ := json.MarshalIndent(register.Token, "", "\t")
+
+				return string(tokenString), nil
+			} else {
+				// This expired token should get overwritten later on a successful request
+				// We won't delete it here so we reduce the number of times we open the file
+				return "", nil
+			}
+		}
+	}
+
+	return "", nil
+}
+
+func updateStore(store []storeRegister, contextName, namespace string, token authenticationv1.TokenRequest) []storeRegister {
+	existed := false
```

Comment:
Looking at this function I'm thinking that storeRegister is more like a map than a list and could reduce the complexity of iterating always to look for the right one. Let me know if I'm wrong

Reply:
Yeah that's true, the thing is that we'd need to merge the `contextName + namespace` into a single string in order to be able to store it as a key. I was afraid of getting into that due to the operator, but I think a `:` should be fine right? like `okteto_cloud:agustin`